src/actions.ts

```diff
@@ -91,84 +96,144 @@ export function listActions(
 		[ActionId.InitBroadcast]: {
 			name: 'Start broadcast test',
 			options: [
+				{
+					type: 'checkbox',
+					label: 'Use variables for broadcast selection',
+					id: 'use_var',
+					default: false
+				},
 				{
 					type: 'dropdown',
 					label: 'Broadcast:',
 					id: 'broadcast_id',
 					choices: [...broadcastEntries, ...broadcastUnfinishedEntries],
 					default: defaultBroadcast,
+					isVisible: (options) => options.use_var == false
+				},
+				{
+					type: 'textinput',
+					label: 'Broadcast',
+					id: 'broadcast_vars',
+					useVariables: true,
+					tooltip: 'Use unfinished_ followed by a number to dynamically select an unfinished broadcast, or the Video ID for static selection',
+					isVisible: (options) => options.use_var == true
 				},
```

Comment:
This is becoming unmanageable. I think you should create a function `createBroadcastChoices` that takes the varying aspects (like the `choices` for the `broadcast_id`), and returns an array with the use_var, broadcast_id and broadcast_vars elements. The function is then called with the correct parameters instead of the three elements:
```suggestion
				...createBroadcastChoices([...broadcastEntries, ...broadcastUnfinishedEntries]),
```